org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -306,32 +310,48 @@ private void updateMarkers(Map<IResource, List<Issue>> issuesPerResource, Analys
       if (failedFiles.contains(resource)) {
         continue;
       }
-      List<Issue> rawIssues = resourceEntry.getValue();
-      try {
-        if (resource instanceof IFile) {
+      if (resource instanceof IFile) {
+        List<Issue> rawIssues = resourceEntry.getValue();
+        IPath path = resource.getFullPath();
+        try {
+          textFileBufferManager.connect(path, LocationKind.IFILE, new NullProgressMonitor());
+          ITextFileBuffer textFileBuffer = textFileBufferManager.getTextFileBuffer(path, LocationKind.IFILE);
```

Comment:
Probably. The `IDocument` is used for calculating absolute offsets from line numbers (and others), so either it has to read (until necessary), or maybe it has an index.

Btw this operation is essentially the same as done in `master`, I just moved things around a bit.